pennylane/math/quantum.py

```diff
@@ -717,7 +717,7 @@ def fidelity(state0, state1, check_state=False, c_dtype="complex128"):
     # Two pure states, squared overlap
     if state1.shape == (len_state1,) and state0.shape == (len_state0,):
         overlap = np.tensordot(state0, np.transpose(np.conj(state1)), axes=1)
-        overlap = np.linalg.norm(overlap) ** 2
+        overlap = np.abs(overlap) ** 2

```

Comment:
We've had issues before with `np.abs` because of the non-differentiability at 0. Can you check if this is the case here too by checking if differentiability is possible through two orthogonal states?

FYI, our solution was to replace `np.abs(overlap) ** 2`, by ```np.real(overlap) ** 2 + np.imag(overlap) ** 2```. The latter is differentiable everywhere.

Reply:
I'm actually not updating it because it caused a similar error to what caused this PR to happen in the first place. In this particular ordering, it says the ArrayBox is not differentiable if I use real+imag, but works fine with abs. I won't question it 😅